packages/svelte/src/internal/server/renderer.js

```diff
@@ -617,6 +642,24 @@ export class Renderer {
 			body
 		};
 	}
+
+	/** @param {[string, unknown][]} serialized */
+	static #hydratable_block(serialized) {
+		let entries = [];
+		for (const [k, v] of serialized) {
+			entries.push(`["${k}",${v}]`);
```

Comment:
I've been struggling with what the correct thing to do here is. Is it to `JSON.stringify` the key, and leave the value to whatever `value.toString` results in? When a user provides `encode`, does that function have to result in a string? Or does it have to result in something that can be `toString`-ed? Or something else entirely?

Reply:
So we're saying "We'll properly encode strings for you -- if you provide a custom encoder we expect it to _emit valid JavaScript_ (which may be a string, in which case it would need to emit its own enclosing quotes)?"